### ISSUE TITLE
AllFailed, AnySucceeded not working

### ISSUE THREAD

**Issue by nwaistd** [argoproj/argo-workflows#4903]
I installed Argo using https://raw.githubusercontent.com/argoproj/argo/stable/manifests/quick-start-postgres.yaml file on minikube on MacBook.

I tested the following workflow using `kubectl apply -f <FILE_PATH>` command.  

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: dag-cycle
  namespace: argo
spec:
  entrypoint: entry
  templates:
  - name: echo
    container:
      image: alpine:3.7
      command: [echo, hello]
  - name: entry
    dag:
      tasks:
      - name: A
        template: echo
        withItems: [ 1, 2 ]
      - name: B
        depends: A.AllFailed
        template: echo
```

And it keeps showing `Failed: invalid spec: templates.entry.tasks.B task result 'All' for task 'A' is invalid` message.

When I changed to `depends: A.AnySucceeded`, it shows `Failed: invalid spec: templates.entry.tasks.B task result 'Any' for task 'A' is invalid`.

I guess this is a parsing problem.

Please help.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
@simster7 is this only available in v2.12 and onwards?

**Comment by simster7** [comment]
This will be a v3 feature, not yet released
